x/ccv/provider/keeper/grpc_query.go

```diff
@@ -221,3 +224,32 @@ func (k Keeper) QueryParams(c context.Context, _ *types.QueryParamsRequest) (*ty
 
 	return &types.QueryParamsResponse{Params: params}, nil
 }
+
+func (k Keeper) QueryOldestUnconfirmVsc(goCtx context.Context, req *types.QueryOldestUnconfirmVscRequest) (*types.QueryOldestUnconfirmVscResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	if req == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "empty request")
+	}
+
+	if req.ChainId == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid request: chain id cannot be empty")
+	}
+
+	if _, consumerRegistered := k.GetConsumerClientId(ctx, req.ChainId); !consumerRegistered {
+		return nil, status.Error(
+			codes.NotFound,
+			errorsmod.Wrap(types.ErrUnknownConsumerChainId, req.ChainId).Error(),
+		)
+	}
+
+	ts, found := k.GetFirstVscSendTimestamp(ctx, req.ChainId)
```

Comment:
ni: Could add a comment here on why `GetFirstVscSendTimestamp` returns the timestamp from the oldest unconfirmed `VSCPacket`, e.g., that everything under the timestamps are getting deleted when we handle `VSCMaturedPacket`s etc.